react-backend/client/src/components/authentication/logincb.js

```diff
@@ -2,6 +2,7 @@ import React, { Component } from 'react';
 import { Link } from 'react-router-dom';
 import { Input } from '../visitors/input';
 import { Button } from '../visitors/button';
+import errorMessages from '../errors';
```

Comment:
you could use destructuring here

```js
import { NO_INPUT, EMAIL_ERROR, DETAILS_ERROR } from '../errors';
```

this would mean rather than typing `errorMessages.NO_INPUT` when you use the error message, you could just type `NO_INPUT`. 

I think the way you've done it is fine, just wanted to point this out as an option.

Reply:
oh right!